Issue with stack[::][::] Mac and Linux

I'm getting a compilation error on this:
```
import numpy as np

#pythran export test(float[::][::], float[::][::], int)
def test(a,b,axis):
    return np.stack((a,b), axis)
```
```
/home/UAUDIO/jlaroche/temp/pythran/pythran/pythonic/include/types/numpy_gexpr.hpp:611:7: error: static assertion failed: should never be instanciated
       static_assert(M == 0, "should never be instanciated");
```
Is this a bug?

--- a/pythran/pythonic/include/types/numpy_gexpr.hpp
+++ b/pythran/pythonic/include/types/numpy_gexpr.hpp
@@ -605,12 +605,6 @@ namespace types
       return _shape;
     }
 
-    template <size_t M>
-    numpy_gexpr reshape(array<long, M> const &shape) const
-    {
-      static_assert(M == 0, "should never be instanciated");
-    }
-
     template <class E>
     long buffer_offset(E const &e, long n) const
     {
@@ -870,6 +864,12 @@ namespace types
     ndarray<dtype, shape_t> copy() const
     {
       return {*this};
+    }
+
+    template <class pS>
+    ndarray<dtype, pS> reshape(pS const &shape) const
+    {
+      return copy().reshape(shape);
     }
   };
 

--- a/pythran/tests/test_numpy_func2.py
+++ b/pythran/tests/test_numpy_func2.py
@@ -228,6 +228,10 @@ def test_copy0(x):
                       2*numpy.ones((4,2,3)),
                       np_stack4=[NDArray[float,:,:,:],NDArray[float,:,:,:]])
 
+    def test_stack5(self):
+        self.run_test("def np_stack5(a): import numpy as np; return np.stack((a[:,1:],a[:,1:]),axis=1)",
+                      numpy.ones((4,3,3)),
+                      np_stack5=[NDArray[float,:,:,:]])
 
     def test_bincount0(self):
         self.run_test("def np_bincount0(a): from numpy import bincount ; return bincount(a)", numpy.arange(5), np_bincount0=[NDArray[int,:]])

